### ISSUE TITLE
Production ready?

### ISSUE THREAD

**Issue by manikmagar** [Afrozaar/wp-api-v2-client-java#7]
Hi, not really an issue but couldn't find any other place to write. I really liked these API. I had started creating my own rest client for wp-api but later discovered this and thought why to reinvent the wheel :).  Thanks for creating these clean and neat java client for WP API.

I am planning to use these API in my J2EE web app which will talk to Wordpress CMS with these. I wanted to thoughts on how stable and production-ready these api's are? I presume these are open source and free to use/modify :). I know these are all new (and so does WP REST API).

Thanks.

**Comment by johanmynhardt** [comment]
Hi @manikmagar :)

I am glad that you found it useful!

Well, taking into consideration that WP-API itself is still in beta (release notes appeared just this morning: https://make.wordpress.org/core/2016/04/04/wp-rest-api-2-0-beta-13-roadmap/), it is certain that there are still some things changing, and with some version changes, there are breaking changes introduced that needs fixing. It is also dependent on the WordPress version.

That said, the features that are needed for our purposes are implemented and working. There are still a number of contexts that need implementing, but it is currently out of scope for our requirements and contributions are welcome.

**Comment by manikmagar** [comment]
Hi @johanmynhardt,

and I am glad, you wrote it :).

Yes, I completely understand WP-API being in beta and per the latest release note hope it gets to stable release 2.X soon.

I have already started using WP-API with this Java Rest client in my Java Web. So far It is doing the job as expected :) and If I dont find any functionality that I need, I will try to add it to client.

Before this, when I was writing my own client, I had used Apache HTTP Client with Apache HTTP Cache to cache the responses for almost static endpoints. That gave me performance boost for pulling static data from Wordpress and make real call remote only once in 24 hours. I see you are using RestTemplate and as RestTemplate does support to use Apache HTTP as connection manager. So I might have to look into that (not immediately as I want to first build it and then look for optimizations). The other option I might look is using Spring Cache to cache method responses, again it has some time.

I will be happy to contribute wherever I can :)

You know what, I was thinking to reach out to WP API Documentation team to include reference of this Java Client on http://v2.wp-api.org/guide/plugins/. This might help all those java community members who are looking for similar solution.

**Comment by manikmagar** [comment]
One more thing, you mentioned using JSON Basic Authentication 0.1. I wanted to mention here that I used this API and Application Password plugin as mentioned on WP API documents - http://v2.wp-api.org/guide/authentication/#application-passwords-or-basic-authentication.

It worked perfectly so far and I think Application Password could be a better option that JSON basic Authentication, my thought.

**Comment by clintonbosch** [comment]
Thanks Manik

This is a great suggestion thank you, we will look to implement this if we get some time, in the meantime any contributions will be great

Thanks again
